ui/lib/services/project_service.dart

```diff
@@ -119,6 +119,14 @@ class ProjectService {
     });
   }
 
+  Future<Map<String, Set<Dependency>>> projectObligation() async {
+    _assertProjectSelected();
+
+    return _execute(() async {
+      return await _client.findObligations(_currentProject!.id);
```

Comment:
The method returns a `Future`, so the combination of `return await` seems odd in an `async` method.

Please learn about Futures in Dart, and then remove the `async` and the `await` in this invocation. And since you are then left with a single method body, you can additionally change it into an "expression body".